spec/optional_query_cache_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'action_controller/metal/exceptions'
+require 'active_record'
+require 'rack'
+
+require 'spec_helper'
+
+require 'optional_query_cache'
+
+RSpec.describe OptionalQueryCache::QueryCache do
+  let(:app) { ->(env) { [200, env, 'app'] } }
+
+  let :subject do
+    described_class.new(app)
+  end
+
+  let :env do
+    env = double
+    allow(env).to receive(:[]).with('REQUEST_PATH').and_return('')
+    env
+  end
```

Comment:
This would be neater as a tap:

```
let :env do
  double.tap |env|
    allow(env).to receive(:[]).with('REQUEST_PATH').and_return('')
  end
end